docs/chymyst01.md

```diff
@@ -386,43 +395,164 @@ Sometimes it appears that repeating input molecules is the most natural way of e
+
+```
+
+There is no nondeterminism in these chemical laws, and no priority needs to be explicitly assigned.
+
+The chemical machine forces the programmer to design the chemistry in such a way that
+the order of running reactions is completely determined by the data on the available molecules.
+
 ## Summary so far
 
 The chemical machine requires for its description:
+
 - a list of defined molecules, together with their types;
 - a list of reactions involving these molecules as inputs, together with reaction bodies.
 
-The user can define reactions in one or more join definitions.
-One join definition encompasses all reactions that have some _input_ molecules in common.
+These definitions comprise the chemical laws of a concurrent program.
 
-After defining the molecules and specifying the reactions, the user can start injecting molecules into the soup.
+The user can define reactions in one or more join definitions.
+Each join definition encompasses all reactions that have some _input_ molecules in common.
+Different join definitions must have no input molecules in common.
 
 In this way, a complicated system of interacting concurrent processes can be specified through a particular set of chemical laws and reaction bodies.
 
+After defining the molecules and specifying the reactions, the user can start the program by injecting some initial molecules into the soup.
+
+The chemical laws fully specify which computations need to be performed for which data.
+
+Let us reiterate the core ideas of the chemical paradigm of concurrency:
+
+In the chemical machine, there is no mutable global state; all data is immutable and must be carried by some molecules.
+Each of these molecules has a specific chemical designation, such as `a`, `b`, `counter`, and so on.
+These chemical designations are not strings `"a"` or `"b"`; one could imagine writing
+
+```scala
+val a = m[Int]
+val q = a
+
+```
+
+This will copy the molecule injector `a` into another local value `q`.
+However, this does not change the chemical designation of a molecule.
+The injector `q` will inject the same molecules as `a`.
+
+The chemical designation of the molecule specifies two aspects of the concurrent program:
+
+- which other input molecules (besides this one) are required to start a computation;
+- which computation will be performed when all the required input molecules are available.
+
+Each reaction specifies its input molecules, and in this way determines all the data necessary for computing the reaction body.
+The chemical machine will automatically make this data available, since a reaction can start only when all its input molecules are present in the soup.
+
+Each reaction also specifies a reaction body, which is a Scala expression that evaluates to `Unit`.
+This expression can perform arbitrary computations using the input molecule values.
+It can also inject new molecules into the soup, which is a side effect of calling a molecule injector.
```

Comment:
I already have a detailed comparison with actor model in the README. Maybe it is good to add a more detailed comparison in a later chapter of the tutorial. I actually feel that the users should not think too much about the actor model when learning this, - the differences can be quite subtle.